circuits/query/query.circom

```diff
@@ -0,0 +1,65 @@
+pragma circom 2.0.0;
+include "../../node_modules/circomlib/circuits/mux3.circom";
+include "../../node_modules/circomlib/circuits/bitify.circom";
+include "../../node_modules/circomlib/circuits/comparators.circom";
+include "comparators.circom";
+
+/*
+  Operators:
+ "0" - equals
+ "1" - less-than
+ "2" - greater-than
+ "3" - in
+ "4" - notin
+*/
+template Query (valueLevels) {
+    // signals
+    signal input in;
+    signal input value[valueLevels];
+    signal input operator;
+    signal output out;
+
+    // operation components
+    component eq = IsEqual();
+    eq.in[0] <== in;
+    eq.in[1] <== value[0];
+
+    // LessThan
+    component lt = LessThan(252);
+    lt.in[0] <== in;
+    lt.in[1] <== value[0];
+
+    component gt = GreaterThan(252);
+    gt.in[0] <== in;
+    gt.in[1] <== value[0];
+
+    // in
+    component inComp = IN(valueLevels);
+    inComp.in <== in;
+    for(var i = 0; i<valueLevels; i++){inComp.value[i] <== value[i];}
+
+    // mux
+    component mux = Mux3();
+    component n2b = Num2Bits(3);
+    n2b.in <== operator;
+
+    n2b.out[0] ==> mux.s[0];
```

Comment:
It's better from readability standpoint to have assignments everywhere from right to left.